snuba/utils/kafka/consumers/abstract.py

```diff
@@ -0,0 +1,30 @@
+from abc import ABC, abstractmethod
+from typing import Any, Callable, Generic, Optional, Sequence, TypeVar
+
+from confluent_kafka import TopicPartition
+
+
+TMessage = TypeVar("TMessage")
+
+
+class Consumer(Generic[TMessage], ABC):
```

Comment:
This is pulled from #478.

Reply:
This API will likely expand in the future due to bringing in `SynchronizedConsumer` as well, per https://github.com/getsentry/snuba/pull/478/files#r326791041.

At that point, it might make sense to split this back out when the dust settles and share that implementation back with Sentry. (We'll have to go through 3 to 2 compatibility obviously.)